src/core/namespace.c

```diff
@@ -131,6 +131,13 @@ static const MountEntry protect_home_read_only_table[] = {
         { "/root",               READONLY,     true  },
 };
 
+/* ProtectHome=tmpfs table */
+static const MountEntry protect_home_tmpfs_table[] = {
+        { "/home",               TMPFS,        true, .read_only = true, .option_const = "mode=0755" },
+        { "/run/user",           TMPFS,        true, .read_only = true, .option_const = "mode=0755" },
+        { "/root",               TMPFS,        true, .read_only = true, .option_const = "mode=0755" },
```

Comment:
hmm, so if the user selects: `ProtectHome=tmpfs + ReadWriteDirectory=/root` then in your current implementation, /root would be mounted over, no?

I think the TMPFS entry in the MountMode structure should most likely move to the very end of the list, so that anything wins over it, and this will make such a /root bind mount work, among others

Reply:
Indeed. Thanks. I will update this.

Reply:
For `/root`, I'd use `mode=0700`.

Reply:
Updated.